scripts/dev/newReleaseBranch.sh

```diff
@@ -0,0 +1,681 @@
+      setNextDevelopmentVersion "${version}-SNAPSHOT"
+    fi
+  fi
+}
+
+setUpstreamVersionsInCommunityPackaging() {
+  local nextCommunityRepoVersion="${1}"
+  local nextEnterpriseShareVersion="${2}"
+
+  if [[ `getCurrentProject` == "acs-community-packaging" ]]
+  then
+    echo "${prefix}    set parent pom <version> ${nextCommunityRepoVersion}"
+    echo "${prefix}    set <dependency.alfresco-community-repo.version> ${nextCommunityRepoVersion}"
+    echo "${prefix}    set <dependency.alfresco-community-share.version> ${nextEnterpriseShareVersion}"
+    # Do not change the <dependency.acs-packaging.version> as we will need a version of the share distribution that
+    # exists, if we want to do a test build. acs-packaging will update it when a release is made.
+    ed -s pom.xml &>${loggingOut} << EOF
+/        <version>.*$/s//        <version>${nextCommunityRepoVersion}<\/version>/
+/.*dependency.alfresco-community-repo.version.*$/s//        <dependency.alfresco-community-repo.version>${nextCommunityRepoVersion}<\/dependency.alfresco-community-repo.version>/
+/.*dependency.alfresco-community-share.version.*$/s//        <dependency.alfresco-community-share.version>${nextEnterpriseShareVersion}<\/dependency.alfresco-community-share.version>/
+wq
+EOF
+  fi
+}
+
+incrementSchema() {
+  local schemaMultiple="${1}"
+
+  if [[ `getCurrentProject` == "alfresco-community-repo" && "${schemaMultiple}" -gt 0 ]]
+  then
+    local schema=`getSchema`
+    schema=`echo "(${schema} / ${schemaMultiple} + 1) * ${schemaMultiple}" | bc`
+    setSchema "${schema}"
+  fi
+}
+
+checkout() {
+  local project="${1}"
+  local branch="${2}"
+
+  echo "${prefix}  Checkout ${project} ${branch}"
+
+  cd "${ROOT_DIR}/${project}/"
+  git fetch                  &>${loggingOut}
+  git checkout "${branch}"   &>${loggingOut}
+}
+
+createBranchFromTag() {
+  local project="${1}"
+  local tag="${2}"
+  local branch="${3}"
+
+  echo "${prefix}  Create ${project} branch ${branch} from tag ${tag}"
+
+  cd "${ROOT_DIR}/${project}/"
+  git fetch                 &>${loggingOut}
+  git checkout "${tag}"     &>${loggingOut}
+  git switch -c "${branch}" &>${loggingOut}
+}
+
+modifyPomVersion() {
+  local version="${1}"
+  local branchType="${2}"
+  local profiles="${3}"
+  local projectType="${4}"
+
+  if [[ "${projectType}" == "Packaging" ]]
+  then
+    if [[ "${branchType}" == "HotFix" ]]
+    then
+      pomVersion="${version}.1"
+    else
+      pomVersion="${version}"
+    fi
+  else
+    local versionMajor=`getVersionMajor "${version}"`
+    if [[ "${versionMajor}" -lt 23 ]]
+    then
+      pomMajor=`getPomMajor`
+      if [[ "${branchType}" == "HotFix" ]]
+      then
+        pomMinor=`getPomMinor`
+        pomMinor=`increment "${pomMinor}"`
+        pomVersion="${pomMajor}.${pomMinor}"
+      else
+        if [[ "${branchType}" == "Master" ]]
+        then
+          # Allow for two service packs before the next release off master
+          pomMajor=`increment "${pomMajor}"`
```

Comment:
I understand the release of master ups the major version, but incrementing major version twice? Is a service pack release based on a major version? I would normally expect it to be based on minor version releases.

Reply:
Added a logger comment.